server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/GroupMappingTest.java

```diff
@@ -0,0 +1,379 @@
+        .when()
+            .put(GroupsRoutes.ROOT_PATH + "/" + GROUP1_ADDRESS + "/" + USER_DOMAIN1)
+        .then()
+            .statusCode(201);
+
+        RestAssured.given()
+            .port(webAdminGuiceProbe.getWebAdminPort())
+        .when()
+            .put(GroupsRoutes.ROOT_PATH + "/" + GROUP2_ADDRESS + "/" + USER_DOMAIN2)
+        .then()
+            .statusCode(201);
+
+        RestAssured.given()
+            .port(webAdminGuiceProbe.getWebAdminPort())
+        .when()
+            .put(GroupsRoutes.ROOT_PATH + "/" + GROUP1_ADDRESS + "/" + GROUP2_ADDRESS)
+        .then()
+            .statusCode(201);
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .sender(new MailAddress(FROM))
+            .recipient(new MailAddress(GROUP2_ADDRESS))
+            .build();
+
+        try (SMTPMessageSender messageSender = SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, DOMAIN1);
+             IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST_IP, IMAP_PORT)) {
+            messageSender.sendMessage(mail);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(messageSender::messageHasBeenSent);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(() -> imapMessageReader.userReceivedMessage(USER_DOMAIN2, PASSWORD));
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(() -> imapMessageReader.userReceivedMessage(USER_DOMAIN1, PASSWORD));
+        }
+    }
+
+    @Test
+    public void messageShouldNotBeSentWhenGroupLoopMapping() throws Exception {
+        RestAssured.given()
+            .port(webAdminGuiceProbe.getWebAdminPort())
+        .when()
+            .put(GroupsRoutes.ROOT_PATH + "/" + GROUP1_ADDRESS + "/" + USER_DOMAIN1)
+        .then()
+            .statusCode(201);
+
+        RestAssured.given()
+            .port(webAdminGuiceProbe.getWebAdminPort())
+        .when()
+            .put(GroupsRoutes.ROOT_PATH + "/" + GROUP1_ADDRESS + "/" + GROUP2_ADDRESS)
+        .then()
+            .statusCode(201);
+
+        RestAssured.given()
+            .port(webAdminGuiceProbe.getWebAdminPort())
+        .when()
+            .put(GroupsRoutes.ROOT_PATH + "/" + GROUP2_ADDRESS + "/" + GROUP1_ADDRESS)
+        .then()
+            .statusCode(201);
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .sender(new MailAddress(FROM))
+            .recipient(new MailAddress(GROUP1_ADDRESS))
+            .build();
+
+        try (SMTPMessageSender messageSender = SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, DOMAIN1);
+            IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST_IP, IMAP_PORT)) {
+            messageSender.sendMessage(mail);
+
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(messageSender::messageHaveNotBeenSent);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(() -> imapMessageReader.userDoesNotReceiveMessage(USER_DOMAIN1, PASSWORD));
```

Comment:
Does the sender receive an error?

(Write another test to shouw the sender receive an email saying there was an error?)

Reply:
I would like to reuse this existing one function and in fact, that function expect IMAP said: BAD FETCH failed. Invalid messageset